File: 0835-image-overlap/0835-image-overlap.py
class Solution:
    def largestOverlap(self, A: List[List[int]], B: List[List[int]]) -> int:
        
        p1, p2 = [], []
        length = len(A)
        
        for i in range(length):
            for j in range(length):
                
                if A[i][j] == 1:
                    p1.append([i, j])
                
                if B[i][j] == 1:
                    p2.append([i, j])
        
        count = {}
        maxOverlap=0
        
        for a in p1:
            for b in p2:
                x = a[0] - b[0]
                y = a[1] - b[1]
                
                s = str(x)+"#"+str(y)
                count[s] = count.get(s,0) + 1;
                
                maxOverlap = max(maxOverlap, count[s])
                
        return maxOverlap
    

File: 0835-image-overlap/0835-image-overlap.java
class Solution {
    public int largestOverlap(int[][] A, int[][] B) {
        
        List<int[]> p1 = new ArrayList<>();
        List<int[]> p2 = new ArrayList<>();
        
        int len = A.length;
        
        for(int i=0; i<len; i++){
            
            for(int j=0; j<len; j++){
                
                if(A[i][j] == 1){
                    p1.add(new int[] {i, j});   
                }

                if(B[i][j] == 1){
                    p2.add(new int[]{i,j});
                }
            }
        }
        
        Map<String, Integer> map = new HashMap<>();
        int ans = 0;
        for(int[] a: p1){
            for(int[] b: p2){
                
                int x = a[0] - b[0];
                int y = a[1] - b[1];
                
                String s = x + "#" + y;
                map.put(s, map.getOrDefault(s, 0)+1);
                ans = Math.max(ans, map.get(s));
            }
        }
        
        return ans;
    }
}